Description: Configure with the assumption that BLAS has 4 byte integers
 This feature test done in configure was written specifically for little
 endian systems, and always detects a false positive for 8 byte integers
 on big endian systems.
 .
 This patch simply deletes the test and forces the result to be 4 bytes,
 which we know is true for the Debian BLAS package on all architectures.
 This is a stopgap measure to fix the build on big endian systems until
 the feature test can be rewritten to work correctly.
Author: Mike Miller <mtmiller@debian.org>
Bug: https://savannah.gnu.org/bugs/?53853
Last-Update: 2018-05-07
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/m4/octave_blas_f77_func.m4
+++ b/m4/octave_blas_f77_func.m4
@@ -153,29 +153,7 @@
 # 8 bytes, we assume it is 4 bytes.
 # FIXME: this may fail with things like -ftrapping-math.
         AC_MSG_CHECKING([BLAS library integer size])
-        AC_RUN_IFELSE(AC_LANG_PROGRAM(,[[
-      integer*8 n
-      integer*4 n4
-      real s,a(1),b(1),sdot
-      a(1) = 1.0
-      b(1) = 1.0
-c Generate -2**33 + 1.  With BLAS compiled to use 64-bit integers, SDOT
-c will return early, setting the result to 0.  With BLAS compiled to use
-c 32-bit integers, this value should be interpreted as 1 and SDOT will
-c return 1.
-      n = 2
-      n = -4 * (n ** 30)
-      n = n + 1
-c Check that our expectation about the type conversion is correct.
-      n4 = n
-      if (n4 .ne. 1) then
-        print *, 'invalid assumption about integer type conversion'
-        stop 2
-      endif
-      s = sdot(n,a,1,b,1)
-      if (s .ne. 0.0) stop 1
-       ]]),[ax_blas_integer_size=8],
-        [ax_blas_integer_size=4])
+        ax_blas_integer_size=4
         AC_MSG_RESULT([$ax_blas_integer_size])
 
         AC_LANG_POP(Fortran 77)
